src/NLog.Targets.Syslog/SyslogTarget.cs

```diff
@@ -75,35 +75,47 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             Task.WhenAll(tasks)
                 .ContinueWith(t =>
                 {
-                    InternalLogger.Debug("[Syslog] Explicit flush completed");
+                    InternalLogger.Debug(t.Exception?.GetBaseException(), "[Syslog] Explicit flush completed");
                     asyncContinuation(t.Exception?.GetBaseException());
-                })
-                .Wait();
```

Comment:
Wait is needed since this is not a Task returning method: removing Wait will cause a regression to people calling Flush

Reply:
No one calls flush directly. If you look closely then you will see the method is actually called `FlushAsync` so it should not be blocking. If you look even closer then you will see that `asyncContinuation` is modelled just like ContinueWith-function. It was invented before Task-library was coined.

Reply:
> No one calls flush directly

I would be cautious with generalizations: see https://github.com/luigiberrettini/NLog.Targets.Syslog/issues/147

I looked closer but, apart from acknowledging the invetion, in .NET the standard is Task

Reply:
Yes the convention with method ending with Async was introduced with Tasks. But still NLog was born before the Task-system was invented.